src/andromede/model/resolve_library.py

```diff
@@ -46,7 +46,7 @@
 
 def resolve_library(
     input_libs: List[InputLibrary], preloaded_libs: Optional[List[Library]] = None
-) -> Library:
+) -> Dict[str, Library]:
```

Comment:
would be even better to create a new container class for the libraries, like:
```python
class LibrariesRepository:
    ...
    def find_model(lib_id: str, model_name: str):
        ...

    def find_fully_qualified_model(fq_model_name: str): 
        ... # also checks validity of the FQ name and split it
```
which would internally have the dictionary

Reply:
Agree, will do it in another PR